coderd/prometheusmetrics/aggregator_test.go

```diff
@@ -606,6 +616,15 @@ func TestLabelsAggregation(t *testing.T) {
 				metricsAggregator.Update(ctx, sc.labels, sc.metrics)
 			}
 
+			// This code makes sure all updates are processed before we start a collection.
+			// We do this by grepping the logs for "update metrics" messages, which are logged
+			// every time an update is processed.
+			//
+			// This is not ideal as the code relies on implementation details (the log message),
+			require.Eventually(t, func() bool {
+				return strings.Count(updateWatch.String(), "update metrics") == len(tc.given)
+			}, testutil.WaitMedium, testutil.IntervalFast)
```

Comment:
This is a really unfortunate fix. There is no way to hook into the timing. Maybe with quartz?

Reply:
This may need to be some kind of buffer protected by a mutex, or perhaps a buffered channel instead?